docs/developers-guide/driver-changelog.md

```diff
@@ -0,0 +1,117 @@
+  escaping disallowed characters), implement this method.
+
+#### Deprecated methods and vars
+
+The following methods and vars are slated for removal in Metabase 0.45.0 unless otherwise noted.
+
+- `metabase.driver/format-custom-field-name` is now unused. Implement `metabase.driver/escape-alias` instead.
+
+- `metabase.driver.sql.query-processor/escape-alias` has been renamed to `metabase.driver/escape-alias`. Everything
+  else is the same. Implementations of the old method will continue to work for now, but please update your drivers to
+  implement the new method since it will be removed in a future release.
+
+- `metabase.driver.sql.query-processor/field-clause->alias` no longer uses the optional parameter `unique-name-fn`.
+  Aliases are now made unique automatically, after escaping them; implement `metabase.driver/escape-alias` if you need
+  to do something special before they are made unique. (Unique aliases are also escaped a second time if you need to
+  do something /really/ special.)
+
+- `metabase.driver.sql.query-processor/field->alias`, which was deprecated in 0.41.0, is now unused in 0.42.0.
+  Implementing this method no longer has any effect. Implement `metabase.driver/escape-alias` instead if you need to
+  do something special; use the [#19610 information](https://github.com/metabase/metabase/pull/19610) if you the alias
+  for one reason or another. This method is still slated for removal in Metabase 0.44.0.
+
+- `metabase.driver.sql.query-processor/*field-options*` is now unused and is no longer bound automatically. Create
+  your own version if you need it for some reason. See our SQL Server driver for an example.
+
+- `metabase.driver.sql.query-processor/*table-alias*` is now unused and is no longer bound automatically. Use or
+  override `:metabase.query-processor.util.add-alias-info/source-table` from the [#19610
+  information](https://github.com/metabase/metabase/pull/19610) instead.
+
+- `metabase.driver.sql.query-processor/*source-query*` is now unused and is no longer bound automatically. Use
+  `metabase.driver.sql.query-processor/*inner-query*` instead, which is always bound even if we are not inside of a
+  source query.
+
+- `metabase.driver.sql.query-processor/field->identifier` is now unused. Implementing this method should no longer be
+  necessary under any circumstances. Override `->honeysql` for `[<driver> :field]` and manipulate the [#19610
+  information](https://github.com/metabase/metabase/pull/19610) if you need to do something special here.
+
+- `metabase.driver.sql.query-processor/prefix-field-alias` is no longer used. Previously, it was made available to
+  give drivers a change to escape automatically generated aliases for joined Fields. This is no longer necessary,
+  because `metabase.driver/escape-alias` is called on automatically generated aliases. Implement
+  `metabase.driver/escape-alias` if you need to do something special.
```

Comment:
```suggestion
- `metabase.driver.sql.query-processor/prefix-field-alias` is no longer used. Previously, it was made available to
  give drivers a chance to escape automatically generated aliases for joined Fields. This is no longer necessary,
  because `metabase.driver/escape-alias` is called on automatically generates aliases. Implement
  `metabase.driver/escape-alias` if you need to do something special.
```